ui/src/screens/Profile/components/Edit.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import styled from "styled-components";
+
+import Button from "../../../components/Button";
+import { Fieldset, Hint, Label, Radio, TextInput } from "../../../components/FormFields";
+import JobsGoWhereApiClient from "../../../shared/services/JobsGoWhereApiClient";
+import { Auth0Profile, FullProfile } from "../types";
+import ProfileImage from "./ProfileImage";
+
+const RECRUITER = "Recruiter";
+const SEEKER = "Seeker";
+
+const TwoCol = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const Col = styled.div`
+  flex: 1;
+  & + & {
+    margin-left: 0.875rem;
+  }
+`;
+
+const RadiosHolder = styled.div`
+  display: flex;
+  align-items: center;
+  .radio-item + .radio-item {
+    margin-left: 2rem;
+  }
+`;
+
+type FormValues = {
+  first_name: string;
+  last_name: string;
+  profile_type: string;
+  email: string;
+  avatar_url: string;
+  headline?: string;
+  company?: string;
+  website?: string;
+};
+
+const onSubmit: SubmitHandler<FormValues> = async (values) => {
+  try {
+    await JobsGoWhereApiClient.post(`${process.env.REACT_APP_API}/profile`, values);
+    window.location.reload();
+  } catch (error) {
+    console.error(error.toJSON());
+    throw error;
+  }
+};
+
+interface ProfileEditProps {
+  profile: Auth0Profile | FullProfile;
+  handleCancelEdit: () => void;
+  newUser?: true;
+}
+
+const Edit: React.FC<ProfileEditProps> = ({ profile, newUser, handleCancelEdit }) => {
+  const { firstName, lastName, email, picture } = profile;
+  const profileType = ("profileType" in profile && profile.profileType) || RECRUITER;
```

Comment:
undefined resolves to false so i think you can just do `profile.profileType || RECRUITER`, same for the others

Reply:
@ng95junwei You are referring to `"profileType" in profile` being unnecessary right?

At this point in the code, `profile` may or may not have the `profileType` property. Hence I need that expression to tell TS that the following code only runs if "profileType" exists. Omitting it causes a TS error 😅

Reply:
TIL! thanks for teaching